File: Ruby/Civitas/lib/casilla_sorpresa.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  class CasillaSorpresa < Casilla
    def initialize(name, mazo)
      super(name)
      @mazo=mazo
    end

    # Método sobrescrito
    def recibe_jugador(actual, todos)
      if(jugador_correcto(actual, todos))
        sorp=@mazo.siguiente
        informe(actual, todos)
        print "#{sorp.to_s} -> Tipo de sorpresa. \n"
        sorp.aplicar_a_jugador(actual, todos)
      end
    end
  
    
    def to_s
      "\n\t DESCRIPCIÓN CASILLA: \n
        \n Nombre: #{nombre}
        \n Tipo: SORPRESA
        \n Sorpresa: #{@mazo.siguiente.to_s}
        "
    end
  end
end

File: Java/P4/Civitas/src/civitas/CasillaSorpresa.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author arturo
 */
public class CasillaSorpresa extends Casilla{
    private mazoSorpresas mazo=new mazoSorpresas();
    
    CasillaSorpresa(mazoSorpresas mazo, String nombre){
        super(nombre);
        this.mazo=mazo;
    }
    
    @Override
    public void recibeJugador(int actual, ArrayList<Jugador> todos){
         if(jugadorCorrecto(actual, todos)==true){
             Sorpresa s=mazo.siguiente();
             informe(actual, todos);
             s.aplicarAJugador(actual, todos);
         }
     }
}
